railties/lib/rails/application.rb

```diff
@@ -232,7 +232,13 @@ def config_for(name, env: Rails.env)
 
       if yaml.exist?
         require "erb"
-        (YAML.load(ERB.new(yaml.read).result) || {})[env] || {}
+        require "json"
+        require "active_support/ordered_options"
+
+        JSON.parse(
```

Comment:
We don't need to use JSON parse to build this object, We can use `ActiveSupport::InheritableOptions` passing a Hash to the constructor.

Reply:
Thank you! I will rework in the next commit then.

Reply:
I tried to implement your suggestion:

```rb
> a = ActiveSupport::InheritableOptions.new(foo: { bar: :baz})
 => {}
> a.class
 => ActiveSupport::InheritableOptions
> a.foo.class
 => Hash 
```
In my previous every nested config parameter was an `ActiveSupport::OrderedOptions` so it was accessible with string/symbol keys or methods. In this case, I will need then to access every nested configuration as an hash with symbol or string keys.

It seems that json conversion is the easiest way to convert a nested Hash into an object.
I would add one more test to explain my point of view, ok?

Reply:
I don't think we should have all the hashes on the config to be `OrderedOptions`. This doesn't match the current implementation of any rails configuration. Only the top level config is a `OrderedOptions` and hashes continue to be hashes.

Reply:
So you would not consider suitable this behaviour https://github.com/rails/rails/pull/33815/commits/bcd3e749d22480ac68121946c8a30b523d4b1eef ?

Reply:
No. Not only in the return of this method. If the reason for this change is to match secret, this doesn't match secrets anymore. Even allowing using methods doesn't match secrets anymore.

Reply:
Ok! I will then `deep_symbolize_keys` after loading the YAML to make it consistent

Reply:
I guess that after the last commit the change has been applied correctly. If it looks good to you, I would squash all commits and I am satisfied with the change implemented :smile: